File: ch01-solid-principles/s02-open-close/ex02-switch-clean/src/main/C#/SwitchClean/DiscountCalculator.cs
using SwitchClean.strategies;
using System;

namespace SwitchClean
{
    public class DiscountCalculator
    {
        public double calculateDiscount(double amount, int month)
        {
            if (!(month >= 1 && month <= 12))
                throw new ArgumentException("There should be a valid month");
            IDiscountStrategy strategy = DiscountStrategyMap.strategyMap[month];
            return strategy.calculateDiscount(amount, month);
        }
	}
}

File: java/ch22-strategy/ex01-discount-strategy/src/main/java/com/example/DiscountCalculator.java
package com.example;

import com.example.strategies.DiscountStrategyMap;

// tag::source[]
public class DiscountCalculator {
    public double calculateDiscount(double amount, int month) {
        IDiscountStrategy strategy = DiscountStrategyMap.get(month); // <1>
        return strategy.calculateDiscount(amount, month);
    }
}
// end::source[]